pkg/webhook/writer.go

```diff
@@ -0,0 +1,91 @@
+package webhook
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/afero"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// WriterOptions specifies the input and output.
+type WriterOptions struct {
+	InputDir       string
+	OutputDir      string
+	PatchOutputDir string
+
+	// inFs is filesystem to be used for reading input
+	inFs afero.Fs
+	// outFs is filesystem to be used for writing out the result
+	outFs afero.Fs
+}
+
+// SetDefaults sets up the default options for RBAC Manifest generator.
+func (o *WriterOptions) SetDefaults() {
+	if o.inFs == nil {
+		o.inFs = afero.NewOsFs()
+	}
+	if o.outFs == nil {
+		o.outFs = afero.NewOsFs()
+	}
+
+	if len(o.InputDir) == 0 {
+		o.InputDir = filepath.Join(".", "pkg", "webhook")
+	}
+	if len(o.OutputDir) == 0 {
+		o.OutputDir = filepath.Join(".", "config", "webhook")
+	}
+	if len(o.PatchOutputDir) == 0 {
+		o.PatchOutputDir = filepath.Join(".", "config", "default")
+	}
+}
+
+// Validate validates the input options.
+func (o *WriterOptions) Validate() error {
+	if _, err := o.inFs.Stat(o.InputDir); err != nil {
+		return fmt.Errorf("invalid input directory '%s' %v", o.InputDir, err)
+	}
+	return nil
+}
+
+func (o *WriterOptions) writeObjectsToDisk(objects ...runtime.Object) error {
```

Comment:
This can probably go in a util pkg. 

I wonder if we should have "k8s/ioutil" pkg which implements k8s object reader/writer :)